File: buildSrc/src/main/java/ru/easydonate/easypayments/gradle/spigotmapper/extension/SpigotMapperExtension.java
package ru.easydonate.easypayments.gradle.spigotmapper.extension;

import lombok.Getter;
import org.gradle.api.Action;
import org.jetbrains.annotations.NotNull;

@Getter
public abstract class SpigotMapperExtension {

    private final @NotNull GameVersionsCollector gameVersionsCollector;

    public SpigotMapperExtension() {
        this.gameVersionsCollector = new GameVersionsCollector();
    }

    public void gameVersions(@NotNull Action<GameVersionsCollector> action) {
        action.execute(gameVersionsCollector);
    }

}

File: buildSrc/src/main/kotlin/ru/easydonate/easypayments/gradle/spigotmapper/extension/SpigotMapperExtension.kt
package ru.easydonate.easypayments.gradle.spigotmapper.extension

import org.gradle.api.Action

abstract class SpigotMapperExtension {

    val minecraftVersionsCollector = MinecraftVersionsCollector()

    fun minecraftVersions(action: Action<in MinecraftVersionsCollector>) {
        action.execute(minecraftVersionsCollector)
    }

}